libr/core/cmd_search.c

```diff
@@ -683,6 +681,13 @@ static int _cb_hit(RSearchKeyword *kw, void *user, ut64 addr) {
 	return true;
 }
 
+static int _cb_hit(RSearchKeyword *kw, void *user, ut64 addr) {
+	struct search_parameters *param = user;
+	const RSearch *search = param->core->search;
+	int klen = kw? kw->keyword_length + (search->mode == R_SEARCH_DELTAKEY): 0;
+	return _cb_hit_sz (kw, klen, user, addr);
+}
+
 static int c = 0;
```

Comment:
globals must die. move them all on top of the file and put a scary warning comment over it

Reply:
Moved it out of global space and put into a struct